pkg/prove/proof.go

```diff
@@ -107,6 +117,49 @@ func txSharePosition(txs types.Txs, txIndex uint64) (startSharePos, endSharePos
 	return startSharePos, endSharePos, nil
 }
 
+// MsgSharesPosition returns the start and end positions for the shares
+// where a given message, referenced by its wrapped pfb transaction, was published at.
+// Note: only supports transactions containing a single message
+func MsgSharesPosition(tx types.Tx) (beginShare uint64, endShare uint64, err error) {
+	unwrappedTx, isMalleated := types.UnmarshalIndexWrapper(tx)
```

Comment:
[nit] consider avoiding the `malleated` terminology given there is no more malleation

```suggestion
	indexWrappedTx, isIndexWrapped := types.UnmarshalIndexWrapper(tx)
```

Reply:
4595dd2